src/Npgsql/NpgsqlParameterCollection.cs

```diff
@@ -121,7 +151,7 @@ public NpgsqlParameter Add(NpgsqlParameter value)
 
             _internalList.Add(value);
             value.Collection = this;
-            InvalidateHashLookups();
+            ReflectListMutationInLookup(value);
```

Comment:
Can pass index here to avoid re-searching the entire list inside ReflectListMutationInLookup

Reply:
Good catch